### ISSUE TITLE
Assimp VERSION 5.2.5 build error

### ISSUE THREAD

**Issue by sergejj180** [assimp/assimp#5021]
I use MSYS2 mingw64 GCC version 12,2,0 

cmake -G "MSYS Makefiles" -D"CMAKE_SYSTEM_PREFIX_PATH:PATH=/mingw64/x86_64-w64-mingw32"

PROJECT(Assimp VERSION 5.2.5)

my compiler setting

CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.

-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: D:/msys64/mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/msys64/mingw64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Shared libraries enabled
-- compiling zlib from sources
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - done
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Enabled importer formats: AMF 3DS AC ASE ASSBIN B3D BVH COLLADA DXF CSM HMP IRRMESH IQM IRR LWO L
WS M3D MD2 MD3 MD5 MDC MDL NFF NDO OFF OBJ OGRE OPENGEX PLY MS3D COB BLEND IFC XGL FBX Q3D Q3BSP RAW
 SIB SMD STL TERRAGEN 3D X X3D GLTF 3MF MMD
-- Disabled importer formats:
-- Enabled exporter formats: OBJ OPENGEX PLY 3DS ASSBIN ASSXML M3D COLLADA FBX STL X X3D GLTF 3MF PB
RT ASSJSON STEP
-- Disabled exporter formats:
-- Treating all warnings as errors (for assimp library only)
-- Configuring done (8.4s)
-- Generating done (0.2s)
-- Build files have been written to: D:/assimp

I received a compilation error

```
[ 54%] Building CXX object code/CMakeFiles/assimp.dir/AssetLib/glTF/glTFImporter.cpp.obj
In file included from D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.h:30,
                 from D:/assimp/code/AssetLib/glTF/glTFAsset.inl:52,
                 from D:/assimp/code/AssetLib/glTF/glTFAsset.h:1011,
                 from D:/assimp/code/AssetLib/glTF/glTFImporter.cpp:45:
In member function 'void o3dgc::IndexedFaceSet<T>::SetNIntAttribute(long unsigned int, long unsigned int) [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodeHeader(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:135:33:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:176:63: error: array subscript 256 is above array bounds of 'long unsigned int [256]' [-Werror=array-bounds]
  176 |                                              m_nIntAttribute[a] = nIntAttribute;
      |                                              ~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodeHeader(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:246:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nIntAttribute'
  246 |         unsigned long                    m_nIntAttribute      [O3DGC_SC3DMC_MAX_NUM_INT_ATTRIBUTES  ];
      |                                          ^~~~~~~~~~~~~~~
In member function 'void o3dgc::IndexedFaceSet<T>::SetNFloatAttribute(long unsigned int, long unsigned int) [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodeHeader(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:117:35:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:171:65: error: array subscript 256 is above array bounds of 'long unsigned int [256]' [-Werror=array-bounds]
  171 |                                              m_nFloatAttribute[a] = nFloatAttribute;
      |                                              ~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodeHeader(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:245:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nFloatAttribute'
  245 |         unsigned long                    m_nFloatAttribute    [O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                          ^~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetIntAttributeDim(long unsigned int) const [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:218:27:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:92:72: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   92 |                                              return m_dimIntAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:248:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_dimIntAttribute'
  248 |         unsigned long                    m_dimIntAttribute    [O3DGC_SC3DMC_MAX_NUM_INT_ATTRIBUTES  ];
      |                                          ^~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetIntAttributeDim(long unsigned int) const [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:218:27:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:92:72: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   92 |                                              return m_dimIntAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:248:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_dimIntAttribute'
  248 |         unsigned long                    m_dimIntAttribute    [O3DGC_SC3DMC_MAX_NUM_INT_ATTRIBUTES  ];
      |                                          ^~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetNIntAttribute(long unsigned int) const [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:218:27:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:61:70: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   61 |                                              return m_nIntAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:246:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nIntAttribute'
  246 |         unsigned long                    m_nIntAttribute      [O3DGC_SC3DMC_MAX_NUM_INT_ATTRIBUTES  ];
      |                                          ^~~~~~~~~~~~~~~
In file included from D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.h:31:
In member function 'long unsigned int o3dgc::SC3DMCEncodeParams::GetFloatAttributeQuantBits(long unsigned int) const',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:201:29:
D:/assimp/contrib/Open3DGC/o3dgcSC3DMCEncodeParams.h:67:74: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   67 |                                        return m_floatAttributeQuantBits[a];
      |                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcSC3DMCEncodeParams.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcSC3DMCEncodeParams.h:129:37: note: while referencing 'o3dgc::SC3DMCEncodeParams::m_floatAttributeQuantBits'
  129 |         unsigned long               m_floatAttributeQuantBits[O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetFloatAttributeDim(long unsigned int) const [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:201:29:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:87:74: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   87 |                                              return m_dimFloatAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:247:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_dimFloatAttribute'
  247 |         unsigned long                    m_dimFloatAttribute  [O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                          ^~~~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetFloatAttributeDim(long unsigned int) const [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:201:29:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:87:74: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   87 |                                              return m_dimFloatAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:247:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_dimFloatAttribute'
  247 |         unsigned long                    m_dimFloatAttribute  [O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                          ^~~~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetNFloatAttribute(long unsigned int) const [with T = short unsigned int]',
    inlined from 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]' at D:/assimp/contrib/Open3DGC/o3dgcSC3DMCDecoder.inl:201:29:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:56:72: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   56 |                                              return m_nFloatAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'o3dgc::O3DGCErrorCode o3dgc::SC3DMCDecoder<T>::DecodePayload(o3dgc::IndexedFaceSet<T>&, const o3dgc::BinaryStream&) [with T = short unsigned int]':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:245:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nFloatAttribute'
  245 |         unsigned long                    m_nFloatAttribute    [O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                          ^~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetNIntAttribute(long unsigned int) const [with T = short unsigned int]',
    inlined from 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)' at D:/assimp/code/AssetLib/glTF/glTFAsset.inl:869:43:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:61:70: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   61 |                                              return m_nIntAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:246:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nIntAttribute'
  246 |         unsigned long                    m_nIntAttribute      [O3DGC_SC3DMC_MAX_NUM_INT_ATTRIBUTES  ];
      |                                          ^~~~~~~~~~~~~~~
In member function 'o3dgc::O3DGCIFSIntAttributeType o3dgc::IndexedFaceSet<T>::GetIntAttributeType(long unsigned int) const [with T = short unsigned int]',
    inlined from 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)' at D:/assimp/code/AssetLib/glTF/glTFAsset.inl:870:40:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:82:73: error: array subscript 256 is above array bounds of 'const o3dgc::O3DGCIFSIntAttributeType [256]' [-Werror=array-bounds]
   82 |                                              return m_typeIntAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:244:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_typeIntAttribute'
  244 |         O3DGCIFSIntAttributeType         m_typeIntAttribute   [O3DGC_SC3DMC_MAX_NUM_INT_ATTRIBUTES  ];
      |                                          ^~~~~~~~~~~~~~~~~~
In member function 'long unsigned int o3dgc::IndexedFaceSet<T>::GetNFloatAttribute(long unsigned int) const [with T = short unsigned int]',
    inlined from 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)' at D:/assimp/code/AssetLib/glTF/glTFAsset.inl:842:45:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:56:72: error: array subscript 256 is above array bounds of 'const long unsigned int [256]' [-Werror=array-bounds]
   56 |                                              return m_nFloatAttribute[a];
      |                                                     ~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:245:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nFloatAttribute'
  245 |         unsigned long                    m_nFloatAttribute    [O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                          ^~~~~~~~~~~~~~~~~
In member function 'void o3dgc::IndexedFaceSet<T>::SetNFloatAttribute(long unsigned int, long unsigned int) [with T = short unsigned int]',
    inlined from 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)' at D:/assimp/code/AssetLib/glTF/glTFAsset.inl:854:39:
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:171:65: error: array subscript 256 is above array bounds of 'long unsigned int [256]' [-Werror=array-bounds]
  171 |                                              m_nFloatAttribute[a] = nFloatAttribute;
      |                                              ~~~~~~~~~~~~~~~~~~~^
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h: In member function 'void glTF::Mesh::Decode_O3DGC(const SCompression_Open3DGC&, glTF::Asset&)':
D:/assimp/contrib/Open3DGC/o3dgcIndexedFaceSet.h:245:42: note: while referencing 'o3dgc::IndexedFaceSet<short unsigned int>::m_nFloatAttribute'
  245 |         unsigned long                    m_nFloatAttribute    [O3DGC_SC3DMC_MAX_NUM_FLOAT_ATTRIBUTES];
      |                                          ^~~~~~~~~~~~~~~~~
cc1plus.exe: all warnings being treated as errors
make[2]: *** [code/CMakeFiles/assimp.dir/build.make:2615: code/CMakeFiles/assimp.dir/AssetLib/glTF/glTFImporter.cpp.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:204: code/CMakeFiles/assimp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by sergejj180** [comment]
I made a few more attempts to compile assimp-5.2.5, as a result of which I found that MSYS2 mingw64 GCC version 12,2,0 can compile ogre3d(13.6.3) + assimp-5.2.5 but cannot compile assimp-5.2.5 separately from ogre3d(13.6.3)

I made another attempt to build assimp-5.2.5 by copying the sources from the ogre3d(13.6.3) folder along with CMakeCache to another location, but this attempt ended in failure

**Comment by kimkulling** [comment]
I made some fixes for 5.2.5. I guess we should just release 5.2.6 or better: 5.3.

**Comment by tonyrewin** [comment]
i've got build error too on osx with puredata extension:

`In file included from /usr/local/Cellar/assimp/5.2.5/include/assimp/scene.h:55:
/usr/local/Cellar/assimp/5.2.5/include/assimp/mesh.h:915:81: error: expected ';' after expression`

**Comment by kimkulling** [comment]
Shall be fixed, Please reopen if the build error is still there.
